### PR TITLE
Corolla hatch port

### PR DESCRIPTION
**Pull request by wocsor**
sorry for the trojan horse porting, 2019 RAV 4 was a prerequisite to the Corolla hatch since they use the same radar. i'll submit a cabana for this car tomorrow! longitudinal and lateral tuning may also need to be done.

### CODE REVIEW

**Comment by wocsor** [comment]
https://community.comma.ai/cabana/?route=5f5afb36036506e4%7C2019-05-14--02-09-54&max=12&url=https%3A%2F%2Fchffrprivate.blob.core.windows.net%2Fchffrprivate3%2Fv2%2F5f5afb36036506e4%2Fdc286c5b33a1b43f487fa08b0c2a5da5_2019-05-14--02-09-54

**Comment by rbiasini** [comment]
https://community.comma.ai/cabana/?route=cdf2f7de565d40ae%7C2019-04-25--03-53-41&max=20&url=https%3A%2F%2Fchffrprivate.blob.core.windows.net%2Fchffrprivate3%2Fv2%2Fcdf2f7de565d40ae%2F57185802db4568c465e9ce4f082a3708_2019-04-25--03-53-41 <-- Rav4

**Comment by rbiasini** [comment]
Nice. looks clean! @wocsor.
Some changes in `carcontroller.py` might be needed though. What about this message?
```
if frame % 100 == 0 and ECU.DSU in self.fake_ecus:
  can_sends.append(create_fcw_command(self.packer, fcw))
```
On your port, this is both being forwarded and sent. I recommend changing panda so it's not forwarded.

**Comment by wocsor** [comment]
I'm wondering about the car's AEB kicking in without the state sending a FCW, but we can block it anyways since that won't stop the car from breaking if it needs to. it's nice that AEB remains completely intact with these cars

**Comment by wocsor** [comment]
updated panda to block 0x411 from camera. also, are these two ports eligible for the fingerprint bounty for store credit?

**Comment by rbiasini** [comment]
definitely eligible for the micro-bounty! Once merged, these are 2.

**Comment by rbiasini** [comment]
About FCW/AEB: ok, instead of blocking it in panda, I agree we should just bypass it in carcontroller if fingerprint is in TSSP2 list.

**Comment by wocsor** [comment]
there we go, now TSSP2 should handle FCW and AEB instead of openpilot

File: selfdrive/car/toyota/interface.py [hunk 1]
```diff
@@ -270,7 +290,7 @@ def update(self, c):
     ret.cruiseState.available = bool(self.CS.main_on)
     ret.cruiseState.speedOffset = 0.
 
-    if self.CP.carFingerprint in [CAR.RAV4H, CAR.HIGHLANDERH, CAR.HIGHLANDER] or self.CP.enableGasInterceptor:
+    if self.CP.carFingerprint in [CAR.RAV4H, CAR.HIGHLANDERH, CAR.HIGHLANDER, CAR.RAV4_2019, CAR.COROLLA_HATCH] or self.CP.enableGasInterceptor:
```
Comment (rbiasini) [inline selfdrive/car/toyota/interface.py]: @wocsor , are those car allowing stop and go without having to press resume?

Comment (wocsor) [reply selfdrive/car/toyota/interface.py]: they are, apparently both do

**Comment by rbiasini** [comment]
Cool, merged this internally. Will be in next release. Thanks @wocsor !

**Pull request closed without merging**